libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -121,3 +143,108 @@ queryDRepStakeDistr nes creds
+queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes =
+  let comStateMembers =
+        csCommitteeCreds $
+          nes ^. nesEpochStateL . esLStateL . lsCertStateL . certVStateL . vsCommitteeStateL
+      comMembers = getCommitteeMembers (queryGovState nes)
+
+      filteredComMembers =
+        if null coldCredsFilter
+          then comMembers
+          else comMembers `Map.restrictKeys` coldCredsFilter
+
+      -- Start from the committee and lookup the hot credential in the Committee State.
+      -- This will miss 'Unrecognized' members (since we're iterating over the "recognized" ones).
+      recognized :: [(Credential 'ColdCommitteeRole (EraCrypto era), CommitteeMemberState (EraCrypto era))]
+      recognized =
+        Map.assocs filteredComMembers >>= \(ck, expirationEpoch) ->
+          let mbHk = Map.lookup ck comStateMembers
+              mbCommitteeMemberState =
+                do
+                  hkStatus <- hotCredStatus mbHk
+                  status <- memberStatus expirationEpoch
+                  let cms =
+                        CommitteeMemberState
+                          hkStatus
+                          status
+                          (Just expirationEpoch)
+                          nextEpochChange
+                  pure (ck, cms)
+           in maybeToList mbCommitteeMemberState
+
+      filteredUnrecognizedMembers =
+        if null coldCredsFilter
+          then Map.difference comStateMembers comMembers
+          else
+            Map.restrictKeys
+              (Map.difference comStateMembers comMembers)
+              coldCredsFilter
+
+      -- Check for Unrecognized members, by starting from the CommitteeState
+      unrecognized :: [(Credential 'ColdCommitteeRole (EraCrypto era), CommitteeMemberState (EraCrypto era))]
+      unrecognized
+        -- we only look for Unrecognized members when it is contained in the filter, or when there is no filtering
+        | Set.null statusFilter || Set.member Unrecognized statusFilter =
+            unrecognized' filteredUnrecognizedMembers
+        | otherwise = []
+
+      unrecognized' ::
+        Map (Credential 'ColdCommitteeRole (EraCrypto era)) (Maybe (Credential 'HotCommitteeRole (EraCrypto era))) ->
+        [(Credential 'ColdCommitteeRole (EraCrypto era), CommitteeMemberState (EraCrypto era))]
+      unrecognized' members =
+        Map.assocs members >>= \(ck, mbHk) ->
+          let mbCommitteeMemberState = do
+                hkStatus <- hotCredStatus (Just mbHk)
+                let cms =
+                      CommitteeMemberState
+                        hkStatus
+                        Unrecognized
+                        Nothing
+                        nextEpochChange
+                pure (ck, cms)
+           in maybeToList mbCommitteeMemberState
+   in CommitteeMembersState
+        { csCommittee = Map.fromList (recognized ++ unrecognized)
+        , csEpochNo = nes ^. nesELL - 1
```

Comment:
I wrote it like this because of [how we create the RatifyEnv in Ledger](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs#L300), by passing it `currentEpoch - 1`, but I suppose, from the perspective of the caller of the Query, the epochNo is the one from the LedgerState.